src/main/java/org/example/RiderAppCLI.java

```diff
@@ -0,0 +1,193 @@
+                    break;
+
+                case "BILL":
+                    rideID = parts[1];
+
+                    double bill = rideService.billRide(rideID);
+                    Ride currentRide = rideRepository.findById(rideID)
+                            .orElseThrow(InvalidRideException::new);
+
+                    System.out.printf("BILL %s %s %.1f\n", rideID, currentRide.getDriver().getDriverID(), bill);
+                    break;
+
+                case "PAY":
+                    rideID = parts[1];
+                    String type = parts[2];
+
+                    PaymentMethodType paymentMethodType = PaymentMethodType.valueOf(type.toUpperCase());
+                    paymentService.setPaymentMethod(paymentMethodType);
+
+                    PaymentDetailsDTO paymentDetails = paymentService.processPayment(rideID);
+                    if (paymentDetails.getPaymentStatus() == PaymentStatus.FAILED) {
+                        System.out.println("LOW_BALANCE");
+                    }
+                    else {
+                        System.out.printf("PAID %s %.1f VIA %s\n", paymentDetails.getReceiverID(), paymentDetails.getAmount(), paymentMethodType);
+                    }
+                    break;
+
+                case "ADD_MONEY":
+                    riderID = parts[1];
+                    float amount = Float.parseFloat(parts[2]);
+
+                    paymentService.setPaymentMethod(PaymentMethodType.WALLET);
+                    WalletPayment wallet = (WalletPayment) paymentService.getPaymentMethod();
+
+                    float balance = wallet.addMoney(riderID, amount);
+                    System.out.println("CURRENT_BALANCE " + riderID + " " + balance);
+                    break;
+
+                case "ADMIN_REMOVE_DRIVER":
+                    driverID = parts[1];
+
+                    if (adminService.removeDriver(driverID)) System.out.println("REMOVED_DRIVER " + driverID);
+                    break;
+
+                case "ADMIN_LIST_DRIVERS":
+                    N = Integer.parseInt(parts[1]);
+
+                    List<DriverDTO> driverDetails = adminService.listNDriverDetails(N);
+                    for (DriverDTO driver: driverDetails) {
+                        System.out.printf("DRIVER_%s (X=%d, Y=%d) RATING %.1f", driver.getDriverID(), driver.getX(), driver.getY(), driver.getRating());
+                        System.out.println();
+                    }
+                    break;
+
+                case "ADMIN_VIEW_DRIVER_EARNINGS":
+                    driverID = parts[1];
+
+                    float earnings = adminService.getDriverEarnings(driverID).getEarnings();
+                    System.out.printf("DRIVER_EARNINGS %s %.1f\n", driverID, earnings);
+                    break;
+
+                default:
+                    break;
+            }
+        } catch (InvalidRideException | InvalidDriverIDException | NoDriversException e) {
```

Comment:
Add one more catch block here to catch all types of runtime exceptions to make sure that the cli tool is failing gracefully. It will also help you debug issues faster and more easily.